src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -698,10 +697,25 @@ impl Coordinator {
                         explain_ctx,
                     })
                 } else {
+                    // If our timestamp determination relies on a timestamp, and we're not already in a
```

Comment:
Bear in mind that ultimately we want this code to live off thread (I have temporary reverted this in `main` for the moment with #24677). Would I be able to bring back the surrounding `mz_ore::task::spawn_blocking(...)` call with this fix here?

Reply:
I think you would be able to move the LIR optimization off the coord thread again because we should maintain the read holds until the peek is complete. I would like to hold off on doing that though until after we cut the release if that's okay!